src/components/ToggleSwitch.js

```diff
@@ -1,89 +1,59 @@
 const React = require('react');
 const Radium = require('radium');
 const StyleConstants = require('../constants/Style');
+const Icon = require('./Icon');
 
 const ToggleSwitch = React.createClass({
   propTypes: {
-    activeColor: React.PropTypes.string,
-    children: React.PropTypes.node,
-    componentStyle: React.PropTypes.oneOfType([
-      React.PropTypes.array,
-      React.PropTypes.object
-    ]),
-    defaultPosition: React.PropTypes.oneOf(['left', 'right']),
-    inactiveColor: React.PropTypes.string,
-    labelStyle: React.PropTypes.oneOfType([
-      React.PropTypes.array,
-      React.PropTypes.object
-    ]),
+    checked: React.PropTypes.bool,
+    checkedByDefault: React.PropTypes.bool,
+    falseIcon: React.PropTypes.element,
     leftLabel: React.PropTypes.string,
     onToggle: React.PropTypes.func,
     rightLabel: React.PropTypes.string,
+    showIcons: React.PropTypes.bool,
     showLabels: React.PropTypes.bool,
-    toggleStyle: React.PropTypes.oneOfType([
-      React.PropTypes.array,
-      React.PropTypes.object
-    ]),
-    trackStyle: React.PropTypes.oneOfType([
-      React.PropTypes.array,
-      React.PropTypes.object
-    ])
+    styles: React.PropTypes.object,
+    trueIcon: React.PropTypes.element
   },
 
   getDefaultProps () {
     return {
-      activeColor: StyleConstants.Colors.PRIMARY,
-      defaultPosition: 'left',
-      inactiveColor: StyleConstants.Colors.FOG,
-      leftLabel: 'On',
+      checkedByDefault: false,
+      leftLabel: 'Off',
       onToggle () {},
-      rightLabel: 'Off',
-      showLabels: true,
-      toggleStyle: {},
-      trackStyle: {}
+      rightLabel: 'On',
+      showLabels: false,
+      showIcons: true,
+      styles: {}
     };
   },
 
   getInitialState () {
+    let checked = false;
+
+    if ('checked' in this.props) {
+      checked = this.props.checked;
+    } else if ('checkedByDefault' in this.props) {
+      checked = this.props.checkedByDefault;
+    }
     return {
-      activePosition: this.props.defaultPosition
+      checked: !!checked
     };
   },
 
-  _handleLeftLabelClick () {
-    const activePosition = 'left';
-
-    this.setState({
-      activePosition
-    });
-
-    this.props.onToggle(activePosition);
-  },
-
-  _handleRightLabelClick () {
-    const activePosition = 'right';
-
-    this.setState({
-      activePosition
-    });
-
-    this.props.onToggle(activePosition);
-  },
-
   _handleToggle (event) {
-    const activePosition = this.state.activePosition === 'left' ? 'right' : 'left';
-
     this.setState({
-      activePosition
+      checked: !this.state.checked
+    }, () => {
+      this.props.onToggle(this.state.checked, event);
```

Comment:
I'm cool with using the callback on setState.  I think it helps clarify that setState is an async operation.  It may not always be required given that you can pass values via argument to another function but I like the idea of setting state and then letting other functions pull from state.  It just seems to be a better pattern.